crates/client/sync/src/eth/src/error.rs

```diff
@@ -0,0 +1,18 @@
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Failed to parse provider URL: {0}")]
+    ProviderUrlParse(#[source] url::ParseError),
```

Comment:
after we remove unnecessary fields, we should remove this as well

Reply:
this particular error type is required in one of the cases